### PR TITLE
Prevent spurious href links

### LINKED ISSUE #889: Spurious links in news items
I think a recent change is causing non-sensical links to be generated to files in NEWS items.

These [news items](https://pkgdown.r-lib.org/dev/news/index.html) (on the dev page) have links to seemingly random rdocumentation pages:

- Can now build sites for older packages that don’t have a Authors@R field (#727).
- build_site() now uses colors even if new_process = TRUE (@jimhester).
- build_home() now uses the content-home.html template when the content is a .Rmd file (@goldingn #787).
- build_home(): a link to the source inst/CITATION was added to the authors page (#714).

*Edit*: Basically anything in backticks is being linked now.

### PR DESCRIPTION
**Pull request by jayhesselberth**
Fixes #889

### CODE REVIEW

**Comment by jayhesselberth** [comment]
This is brittle but is the best I could come up with.

**Comment by jayhesselberth** [comment]
But I don't think this fixes #895?

**Comment by jayhesselberth** [comment]
Are there any other operators we should exclude?

https://stat.ethz.ch/R-manual/R-devel/library/base/html/Syntax.html

**Comment by hadley** [comment]
I think we should only link regular prefix functions; I know there is a code path that only activates linking if there's an explicit `()`; we should probably do that for base R functions.

**Comment by hadley** [comment]
Oh it's `bare_symbol` I was thinking of, but that's not quite what we want.

I think we should just systematically excluded all infix operators. We're already excluding custom infix operators above, so it's just a matter of constructing a reasonably comprehensive list from `?Syntax`.

File: R/link-href.R [hunk 1]
```diff
@@ -29,6 +29,12 @@ href_expr <- function(expr, bare_symbol = FALSE) {
     if (grepl("^%.*%$", fun_name))
       return(NA_character_)
 
+    # prevent spurious links to Authors@R, content-home.html,
+    # toc: depth, etc.
+    # Black list `@`, `:`, and arithmetic operators
+    if (fun_name %in% c("@", "+", "-", "/", "*", "=", ":"))
```
Comment (hadley) [inline R/link-href.R]: I think you could just make a single comment for this block and the block above: `# Exclude infix functions`.  Or maybe extract out in `is_infix()` function, and then we wouldn't even need the comment.

Comment (hadley) [reply R/link-href.R]: Actually we should assert positively that it's a prefix function, also excluding the special forms:

```R
is_prefix <- function(fun) {
  if (grepl("^%.*%$", fun_name)) {
    return(FALSE)
  }
  
  infix <- c(
    "::", ":::", "$", "@", "[", "[[", "^", "-", "+", ":", "<",
    ">", "<=", ">=", "==", "!=", "!", "&", "&&", "|", "||", "~",
    "->", "->>", "<-", "<<-", "=", "?"
  )
  if (fun %in% infix) {
    return(FALSE)
  }
  
  special <- c(
    "(", "{", "if", "for", "while", "repeat", "next", "break", "function"
  )
  if (fun %in% infix) {
    return(FALSE)
  }
  
  TRUE
}
```

**Pull request merged**